src/pip/_internal/req/req_install.py

```diff
@@ -565,31 +577,78 @@ def pyproject_toml(self):
+        build-backend) and discard the rest.
         """
-        # If pyproject.toml does not exist, don't do anything.
-        if not os.path.isfile(self.pyproject_toml):
-            return None
+        if self._pyproject_toml_loaded:
+            return
+
+        # Don't do this processing twice
+        self._pyproject_toml_loaded = True
+
+        has_pyproject = os.path.isfile(self.pyproject_toml)
+        has_setup = os.path.isfile(self.setup_py)
+
+        if has_pyproject:
+            with io.open(self.pyproject_toml, encoding="utf-8") as f:
+                pp_toml = pytoml.load(f)
+            build_system = pp_toml.get("build-system")
+        else:
+            build_system = None
+
+        # The following cases must use PEP 517
+        # We check for use_pep517 equalling False because that
+        # means the user explicitly requested --no-use-pep517
+        if (has_pyproject and not has_setup):
+            if self.use_pep517 is False:
+                raise InstallationError(
+                    "Disabling PEP 517 processing is invalid: "
+                    "project does not have a setup.py"
+                )
+            self.use_pep517 = True
+        if (build_system and "build-backend" in build_system):
+            if self.use_pep517 is False:
+                raise InstallationError(
+                    "Disabling PEP 517 processing is invalid: "
+                    "project specifies a build backend of {} "
+                    "in pyproject.toml".format(
+                        build_system["build-backend"]
+                    )
+                )
+            self.use_pep517 = True
+
+        # Choose whether to use PEP 517 if the user didn't say
+        if self.use_pep517 is None:
+            if has_pyproject:
+                self.use_pep517 = True
+            else:
+                self.use_pep517 = False
+
+        if build_system is None:
+            if self.use_pep517:
+                build_system = {
+                    "requires": ["setuptools>=38.2.5", "wheel"],
+                    "build-backend": "setuptools.build_meta",
+                }
+            else:
+                build_system = {
+                    "requires": ["setuptools", "wheel"],
+                }
+
+        assert self.use_pep517 is not None
+        assert build_system is not None
 
         error_template = (
             "{package} has a pyproject.toml file that does not comply "
             "with PEP 518: {reason}"
         )
 
-        with io.open(self.pyproject_toml, encoding="utf-8") as f:
-            pp_toml = pytoml.load(f)
-
-        # If there is no build-system table, just use setuptools and wheel.
-        if "build-system" not in pp_toml:
-            return ["setuptools", "wheel"]
-
-        # Specifying the build-system table but not the requires key is invalid
-        build_system = pp_toml["build-system"]
+        # Specifying the build-system table but not
```

Comment:
this comment wrapping seems unnecessary

Reply:
Yeah, I was probably over-reacting to flake8 complaints...